in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+        B ->> C: Gossip Attestations
+    end
+
+    A ->> E: Propose Block
+    D ->> E: Retrieve TxHash
+    par
+        D ->> B: Request TxEffect + Proofs
+        D ->> C: Request TxEffect + Proofs
+    end
+
+    D ->> D: Execute block and check state root on L1
+    
+    alt If state root matches L1
+        D ->> D: Do nothing
+    else If state root does not match L1
+        D ->> C: Send proposal to slash malicious signatures
+    end
+
+    C ->> E: Retrieve Tx Hash
+    C ->> D: Request TxEffect + Proofs
+    C ->> C: Execute Proposal
+    alt If state root matches L1
+        C ->> C: Do nothing
+    else If state root doesn't match L1
+        C ->> D: Sign slashing proposal
+    end
+    D ->> D: Aggregate BLS signatures
+    D ->> E: Send BLS signatures and aggregated public key
+```
+
+**What should the validators do if they can't obtain the necessary data to re-execute the invalid state root?** 
+
+The above scenario describes the "happy path" where a stupid committee coordinates an attack and shares the data with the rest of the validator set. In a sophisticated attack, the committee will plot to withhold the data. 
+
+The question boils down to: Do we punish data withholding attacks given that they cannot be verified by the L1? 
+
+The answer is -> Yes! A majority stake voting incorrectly on a data withholding attack is an even stronger violation of our honest-committee assumption. Honest committee members protect themselves by gossipping the tx data to as many peers possible. 
+
+Therefore the Executing Validators who can't download all `TxObject` and proof data for any given block should initiate a data withholding slashing proposal, which works in exactly the same way as the invalid state root slashing proposal.
+
+
+### 2. ETH Congestion
+
+Provers may not be able to post proofs if the L1 is congested or is experiencing an inactivity leak. Instead of automatic slashing of the prover bond, the validator set can vote to "unslash" the provers. 
+
+The prover's bond sits in escrow (or a separate contract) for $t=4$ weeks. During this period, the prover can plead their case and convince the validators to vote to unslash them. A validator must initiate a vote to unslash the prover and submit to L1 for verification on time, otherwise the bond gets slashed. 
+
+Since "Ethereum was congested" is a subjective statement, two ways we could implement this:
+
+1) Make $t$ larger, giving the prover enough time to convince the validators to manually update their software to vote to unslash the prover. This could be done on the forums or validator community channels for example. 
+2) Add an environment variable `max_eth_base_fee` that clients agree to implement. If `base_gas_fee > max_gas_fee` at the proof submission deadline time, then validator nodes automatically sign votes to unslash prover.
+
+They are largely equivalent except that the second requires less coordination.
+
+### PoG Non-slashable offences
```

Comment:
If you loosen/simplify the generic slashing function on the rollup, these are all slashable.

I think the goal should be that one is able to post on a forum and say "so and so did X", circulate a slashing proposal via p2p, get the votes via p2p, and effect the consequences.